cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectAssemblyOperatorConnectorRestartTest.java

```diff
@@ -0,0 +1,250 @@
+    }
+
+    @ParameterizedTest
+    @MethodSource("annotationValues")
+    public void testRestartAnnotationValueHasRestart(String annotationValue, boolean shouldRestart, boolean shouldBeValid) {
+        KafkaConnectAssemblyOperator op = new KafkaConnectAssemblyOperator(null, new PlatformFeaturesAvailability(true, KubernetesVersion.MINIMAL_SUPPORTED_VERSION),
+                ResourceUtils.supplierWithMocks(false), ResourceUtils.dummyClusterOperatorConfig());
+        HasMetadata resource = new KafkaConnectBuilder()
+            .withNewMetadata()
+            .withAnnotations(Map.of(Annotations.ANNO_STRIMZI_IO_RESTART, annotationValue))
+            .endMetadata()
+            .build();
+        boolean hasRestart = op.hasRestartAnnotation(resource, null);
+        assertThat(String.format("value '%s' is expected to %s", annotationValue, shouldRestart ? "restart" : "NOT restart"), hasRestart, equalTo(shouldRestart));
+    }
+
+    private static Stream<Arguments> annotationValues() {
+        return Stream.of(
+                arguments("true", true, true),
+                arguments("includeTasks,onlyFailed", true, true),
+                arguments("onlyFailed,includeTasks", true, true),
+                arguments("    onlyFailed   ,   includeTasks   ", true, true),
+                arguments("includeTasks", true, true),
+                arguments("onlyFailed", true, true),
+                arguments("   onlyFailed   ", true, true),
+                arguments("    includeTasks   ", true, true),
+                arguments("false", false, false),
+                arguments("true,includeTasks", true, false),
+                arguments("true,includeTasks,onlyFailed", true, false),
+                arguments("true,onlyFailed", true, false),
+                arguments("true,invalidArg,onlyFailed", true, false),
+                arguments("invalidArg,onlyFailed", true, false),
+                arguments("onlyFailed,invalidArg", true, false),
+                arguments("includeTasks,invalidArg", true, false),
+                arguments("invalidArg,includeTasks", true, false),
+                arguments("", false, false),
+                arguments("     ", false, false),
+                arguments("   b  ", true, false)
+        );
+    }
+
+    private KafkaConnector buildKafkaConnector(String restartAnnotationValue) {
+        return new KafkaConnectorBuilder()
+                .withNewMetadata()
+                .withName("my-connector")
+                .withNamespace("my-namespace")
+                .withAnnotations(Map.of(Annotations.ANNO_STRIMZI_IO_RESTART, restartAnnotationValue))
+                .endMetadata()
+                .withNewSpec()
+                .withClassName("MyClass")
+                .withTasksMax(3)
+                .withConfig(Map.of("topic", "my-topic"))
+                .endSpec()
+                .build();
```

Comment:
Tiny thing, but we usually indent these for improved readability
```suggestion
                .withNewMetadata()
                    .withName("my-connector")
                    .withNamespace("my-namespace")
                    .withAnnotations(Map.of(Annotations.ANNO_STRIMZI_IO_RESTART, restartAnnotationValue))
                .endMetadata()
                .withNewSpec()
                    .withClassName("MyClass")
                    .withTasksMax(3)
                    .withConfig(Map.of("topic", "my-topic"))
                .endSpec()
                .build();
```